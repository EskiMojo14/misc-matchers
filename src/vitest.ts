/* eslint-disable @typescript-eslint/no-empty-interface */
import { expect } from "vitest";
import * as mixNMatchers from ".";

declare module "vitest" {
  interface Assertion<T> extends mixNMatchers.MixNMatchers<void, T> {}
  interface AsymmetricMatchersContaining
    extends mixNMatchers.AsymmetricMixNMatchers {
    enum: mixNMatchers.AsymmetricMixNMatchers["ofEnum"];
  }
}

expect.extend(mixNMatchers);
expect.extend({ enum: mixNMatchers.ofEnum });
