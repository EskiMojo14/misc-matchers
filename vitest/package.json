{
  "name": "mix-n-matchers-vitest",
  "version": "1.0.0",
  "description": "Miscellaneous custom Jest matchers",
  "type": "module",
  "main": "../dist/vitest.cjs",
  "module": "../dist/vitest.js",
  "types": "../dist/vitest.d.ts",
  "files": [
    "../dist"
  ]
}
