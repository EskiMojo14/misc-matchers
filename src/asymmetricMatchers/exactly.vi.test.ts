import { expect, it, describe, vi } from "vitest";

describe("exactly", () => {
  it("allows Object.is equality where deep equality would normally be used", () => {
    const ref = {};
    const fn = vi.fn();
    fn(ref);
    expect(fn).toHaveBeenCalledWith(expect.exactly(ref));
    expect(() => {
      expect(fn).toHaveBeenCalledWith(expect.exactly({}));
    }).toThrowErrorMatchingSnapshot();

    expect(fn).toHaveBeenCalledWith(expect.not.exactly({}));
    expect(() => {
      expect(fn).toHaveBeenCalledWith(expect.not.exactly(ref));
    }).toThrowErrorMatchingSnapshot();
  });
});
