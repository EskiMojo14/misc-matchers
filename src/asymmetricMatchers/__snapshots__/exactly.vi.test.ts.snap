// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`exactly > allows Object.is equality where deep equality would normally be used 1`] = `
expected "spy" to be called with arguments: [ CustomMatcher{ â€¦(3) } ][90m

Received: 

<b>  1st spy call:

</><d>  Array [</>
<g>-   exactly<[object Object]>,[90m
<r>+   Object {},[90m
<d>  ]</>
</>[90m

Number of calls: <b>1</>
</>
`;

exports[`exactly > allows Object.is equality where deep equality would normally be used 2`] = `
expected "spy" to be called with arguments: [ CustomMatcher{ â€¦(3) } ][90m

Received: 

<b>  1st spy call:

</><d>  Array [</>
<g>-   not.exactly<[object Object]>,[90m
<r>+   Object {},[90m
<d>  ]</>
</>[90m

Number of calls: <b>1</>
</>
`;

exports[`exactly allows Object.is equality where deep equality would normally be used 1`] = `
<d>expect(</><r>jest.fn()</><d>).</>toHaveBeenCalledWith<d>(</><g>...expected</><d>)</>

Expected: <g>exactly<[object Object]></>
Received: <r>{}</>

Number of calls: <r>1</>
`;

exports[`exactly allows Object.is equality where deep equality would normally be used 2`] = `
<d>expect(</><r>jest.fn()</><d>).</>toHaveBeenCalledWith<d>(</><g>...expected</><d>)</>

Expected: <g>not.exactly<[object Object]></>
Received: <r>{}</>

Number of calls: <r>1</>
`;
